src/sif/autocost.cc

```diff
@@ -556,6 +556,7 @@ void ParseAutoCostOptions(const rapidjson::Document& doc,
                           const std::string& costing_options_key,
                           CostingOptions* pbf_costing_options) {
   pbf_costing_options->set_costing(Costing::auto_);
+  pbf_costing_options->set_name(Costing_Enum_Name(pbf_costing_options->costing()));
```

Comment:
every costing does this now, uses its enum to get a string name as a default costing name. since `ParseSharedCostOptions` is called after this, the default here will be overriden if the user has set a name themselves. all the other changes in costing files are the same as this one